app/jobs/update_job.rb

```diff
@@ -22,25 +23,39 @@ def perform(model_params:, signed_ids:, background_job_result:)
 
     object_client = Dor::Services::Client.object(model.externalIdentifier)
     existing = object_client.find
-    if existing.version >= model.version
-      background_job_result.output = {
-        errors: [
-          title: 'Version conflict',
-          detail: "The repository already has a version '#{existing.version}' for " \
-                  "#{existing.externalIdentifier}, and you provided '#{model.version}'"
-        ]
-      }
-      background_job_result.complete!
-      return
+
+    if Workflow.open_for_versioning?(model.externalIdentifier, existing.version)
+      unless existing.version == model.version
+        error_title = 'Version update conflict'
+        error_detail = "The repository is on version '#{existing.version}' for " \
+                       "#{existing.externalIdentifier}, which is currently open for update. " \
+                       "You provided '#{model.version}', but must update the open version."
+        complete_job_with_error!(error_title, error_detail, background_job_result)
+        return
+      end
+    else
+      unless existing.version + 1 == model.version
+        error_title = 'Version addition conflict'
+        error_detail = "The repository is on version '#{existing.version}' for " \
+                       "#{existing.externalIdentifier}, which is closed for update. You provided " \
+                       "'#{model.version}', but must provide the subsequent version, which will be opened."
+        complete_job_with_error!(error_title, error_detail, background_job_result)
+        return
+      end
+
+      object_client.version.open
     end
 
-    object_client.version.open
     object_client.update(params: model)
 
     background_job_result.output = { druid: model.externalIdentifier }
 
     StageFiles.stage(signed_ids, model.externalIdentifier) do
-      object_client.version.close(description: 'Update via sdr-api', significance: 'major')
+      if start_workflow
+        Workflow.create_unless_exists(model.externalIdentifier, 'accessionWF', version: existing.version)
```

Comment:
i wasn't 100% sure `existing.version` was right (as opposed to `model.version`), but i think this is consistent with how `IngestJob` makes the call.

Reply:
And then get rid of the `version.open` above since this will be handled automatically by that `accession.start` call if needed

Reply:
thanks for the pairing on this, @peetucket!  dor-services-client/dor-services-app accessioning did the trick.

Reply:
fwiw, i did end up keeping `version.open` for the case where a new version is being updated, but no accessioning has done.

Reply:
Not sure if it matters, but one thing to note is that we don't kick off accessionWF after opening and closing a version, you won't get a publish.  This means that if the use case for this is a metadata only change, the PURL won't be updated.  My understanding was always that a version with accessioning was needed for any changes that manifest on the PURL for this reason.